certs/sp/demo_sp.crt

```diff
@@ -0,0 +1,22 @@
+-----BEGIN CERTIFICATE-----
```

Comment:
this is the x509 cert -- we use a fingerprint of this cert so that the "service provider" (in this case, the identity idp app) knows who is making the request. 

the way one SO responder but it:

> You can use the public key to verify that the content of the SAML response matches the key - in other words - that response definitely came from someone who has the matching private key to the public key in the message, and the response hasn't been tampered with.

this is the public key. :)

more info here: https://developers.onelogin.com/saml/ruby

also, I may be messing this all up. but this is my working understanding. still learning myself!